pystiche/papers/ulyanov_et_al_2016/data.py

```diff
@@ -0,0 +1,269 @@
+from typing import Optional, Sized
+from urllib.parse import urljoin
+import torch
+from torch.utils.data import Dataset, Sampler, DataLoader
+from pystiche.data import (
+    DownloadableImage,
+    DownloadableImageCollection,
+    CreativeCommonsLicense,
+    PublicDomainLicense,
+    ImageFolderDataset,
+    FiniteCycleBatchSampler,
+)
+from pystiche.image import extract_num_channels
+from pystiche.image.transforms import Transform, Resize, ComposedTransform, CenterCrop, ValidRandomCrop
+from pystiche.image.transforms.functional import grayscale_to_fakegrayscale
+
+
+__all__ = [
+    "ulyanov_et_al_2016_content_transform",
+    "ulyanov_et_al_2016_style_transform",
+    "ulyanov_et_al_2016_images",
+    "ulyanov_et_al_2016_dataset",
+    "ulyanov_et_al_2016_batch_sampler",
+    "ulyanov_et_al_2016_image_loader",
+]
+
+
+def ulyanov_et_al_2016_content_transform(
+    edge_size: int = 256, impl_params: bool = True, instance_norm: bool = True,
+) -> ComposedTransform:
+    class OptionalGrayscaleToFakegrayscale(Transform):
+        def forward(self, input_image: torch.Tensor) -> torch.Tensor:
+            is_grayscale = extract_num_channels(input_image) == 1
+            if is_grayscale:
+                return grayscale_to_fakegrayscale(input_image)
+            else:
+                return input_image
+
+    transforms = []
+    if impl_params:
+        if instance_norm:
+            transforms.append(ValidRandomCrop(edge_size))
+        else:
+            transforms.append(
+                Resize((edge_size, edge_size), interpolation_mode="bilinear")
+            )
+    else:
+        transforms.append(CenterCrop(edge_size))
+
+    transforms.append(OptionalGrayscaleToFakegrayscale())
+    return ComposedTransform(*transforms)
+
+
+def ulyanov_et_al_2016_style_transform(
+    impl_params: bool = True,
+    instance_norm: bool = True,
+    edge_size: Optional[int] = None,
+) -> ComposedTransform:
+    if edge_size is None:
+        edge_size = 256
+
+    if impl_params and instance_norm:
+        interpolation_mode = "bicubic"
+    else:
+        interpolation_mode = "bilinear"
```

Comment:
We can shorten this even more with

```python
interpolation_mode = "bicubic" if impl_params and instance_norm "bilinear"
```

I think this is still legible. Do you agree?